src/main/java/laddergame/domain/LadderGame.java

```diff
@@ -0,0 +1,62 @@
+package laddergame.domain;
+
+import laddergame.domain.gameelements.*;
+import laddergame.domain.ladder.Ladder;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LadderGame {
+    private final Players players;
+    private final Ladder ladder;
+    private final Prizes prizes;
+    private final Map<Player, Prize> playerGameResult;
+
+    public LadderGame(Players players, Ladder ladder, Prizes prizes) {
+
+        this.players = players;
+        this.ladder = ladder;
+        this.prizes = prizes;
+
+        playerGameResult = new LinkedHashMap<>();
+    }
+
+    public void playGame() {
+        ladder.climb(players);
+
+        for (Player player : players.getPlayers()) {
+            Position playerPosition = player.getPlayerPosition();
+            Prize prize = findPrizeByPosition(playerPosition);
+
+            playerGameResult.put(player, prize);
+        }
+    }
+
+    public Prize findPrizeByPlayerName(String playerName) {
+        if(playerGameResult.isEmpty()){
+            throw new IllegalStateException("아직 사다리 게임이 실행되지 않았습니다.");
+        }
+
+        return playerGameResult.get(findPlayerByName(playerName));
+    }
+
+    private Prize findPrizeByPosition(Position playerPosition) {
+        return prizes.getPrizes()
+                .stream()
+                .filter(prize -> prize.getPosition().isSame(playerPosition))
```

Comment:
```suggestion
                .filter(prize -> prize.isSamePosition(playerPosition)) // 적절한 메서드명으로 바꿀 것
```

객체지향 생활체조
4. 한 줄에 점을 하나만 찍는다.